client/signup/processing-screen/index.jsx

```diff
@@ -369,4 +370,5 @@ export class SignupProcessingScreen extends Component {
 
 export default connect( state => ( {
 	useOAuth2Layout: showOAuth2Layout( state ),
+	user: getCurrentUser( state ) || false,
```

Comment:
Can we remove the cast `null` to `false`? The code that does a `this.props.user && this.props.user.email` check will continue to work equally well. The proptype definition will need to be updated, too.